redis.ts

```diff
@@ -22,70 +16,80 @@ import {
   Raw,
   BulkNil,
 } from "./command.ts";
+import { RedisConnection } from "./connection.ts";
 
 export type Redis = RedisCommands & {
   executor: CommandExecutor;
 };
 
 class RedisImpl implements RedisCommands {
-  _isClosed = false;
+
   get isClosed() {
-    return this._isClosed;
+    return this.connection?.isClosed;
+  }
+
+  get isConnected() {
+    return this.connection?.isConnected;
+  }
+
+  get executor() {
+    return this.connection?.executor!;
   }
 
   constructor(
-    private closer: Closer,
-    private writer: BufWriter,
-    private reader: BufReader,
-    readonly executor: CommandExecutor,
+    private connection: RedisConnection,
   ) {
   }
 
+  close() {
+    return this.connection?.close();
+  }
+
   async execStatusReply(
     command: string,
     ...args: (string | number)[]
   ): Promise<Status> {
-    const [_, reply] = await this.executor.exec(command, ...args);
+    const [_, reply] = await this.connection?.executor!.exec(command, ...args);
```

Comment:
I think it's better to define `RedisConnection.exec` than referencing `RedisConnection.executor`.

```suggestion
    const [_, reply] = await this.connection?.exec(command, ...args);
```